src/Sema.zig

```diff
@@ -17133,6 +17133,15 @@ fn zirPtrCast(sema: *Sema, block: *Block, inst: Zir.Inst.Index) CompileError!Air
     else
         operand;
 
+    if (!dest_ty.isPtrLikeOptional() and operand_ty.isPtrLikeOptional() and
+        try sema.isComptimeKnown(block, operand_src, operand))
+    {
+        const operand_val = try sema.resolveConstValue(block, operand_src, operand, "use of undefined value here causes undefined behavior");
```

Comment:
The `reason` param is supposed to be the reason why the value has to be comptime known, "use of undefined value here causes undefined behavior" makes no sense as the reason. Also you just checked that the value is comptime known so this reason should never be used in the first place.

It would be better to just directly use `resolveMaybeUndefVal`.

Reply:
Right, didn't understand that. In that case, should it just be unwrapped like `(try sema.resolveMaybeUndefVal(block, operand_src, operand)).?` ?

Reply:
Sure, but it'd be even better to just assign that result to a constant and avoid `isComptimeKnown` all together.

Reply:
Scratch that, I see what you meant now.